File: repos_estudiantes_ALVAREZ_tercera_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/PERSISTENCIADEOBJETOS/EJERCICIO3/Ej3/src/Ej3/ArchivoCliente.java
package Ej3;
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class ArchivoCliente {
    private String nombreArchivo;

    public ArchivoCliente(String nombreArchivo) {
        this.nombreArchivo = nombreArchivo;
    }

    public void crearArchivo() {
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nombreArchivo))) {
            oos.writeObject(new ArrayList<Cliente>());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void guardaCliente(Cliente cliente) {
        List<Cliente> lista = leerLista();
        lista.add(cliente);
        escribirLista(lista);
        System.out.println("Grabado: " + cliente);
    }

    public Cliente buscarCliente(int id) {
        List<Cliente> lista = leerLista();
        for (Cliente c : lista) {
            if (c.getId() == id) {
                return c;
            }
        }
        return null;
    }

    public Cliente buscarCelularCliente(int telefono) {
        List<Cliente> lista = leerLista();
        for (Cliente c : lista) {
            if (c.getTelefono() == telefono) {
                return c;
            }
        }
        return null;
    }

    @SuppressWarnings("unchecked")
    private List<Cliente> leerLista() {
        File archivo = new File(nombreArchivo);
        if (!archivo.exists()) {
            return new ArrayList<>();
        }
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nombreArchivo))) {
            return (List<Cliente>) ois.readObject();
        } catch (IOException | ClassNotFoundException e) {
            e.printStackTrace();
            return new ArrayList<>();
        }
    }

    private void escribirLista(List<Cliente> lista) {
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nombreArchivo))) {
            oos.writeObject(lista);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/PERSISTENCIADEOBJETOS/EJERCICIO3/Ej3-PYTHON/ArchivoCliente.py
import json
import os
from Cliente import Cliente

class ArchivoCliente:
    def __init__(self, nomA):
        self.__nomA = nomA

    def crearArchivo(self):
        with open(self.__nomA, 'w') as f:
            json.dump([], f)

    def guardaCliente(self, cliente):
        lista = self.__leer_lista()
        lista.append(cliente.to_dict())
        self.__escribir_lista(lista)

    def buscarCliente(self, id):
        lista = self.__leer_lista()
        for c in lista:
            if c["id"] == id:
                return Cliente.from_dict(c)
        return None

    def buscarCelular(self, numero):
        lista = self.__leer_lista()
        for c in lista:
            if c["telefono"] == numero:
                return Cliente.from_dict(c)
        return None

    def __leer_lista(self):
        if os.path.exists(self.__nomA):
            with open(self.__nomA, 'r') as f:
                return json.load(f)
        return []

    def __escribir_lista(self, lista):
        with open(self.__nomA, 'w') as f:
            json.dump(lista, f, indent=4)